crates/wallet/src/wallet/mod.rs

```diff
@@ -1555,7 +1555,7 @@ impl Wallet {
         };
 
         // sort input/outputs according to the chosen algorithm
-        params.ordering.sort_tx_with_aux_rand(&mut tx, rng);
+        params.ordering.clone().sort_tx_with_aux_rand(&mut tx, rng);
 
```

Comment:
Forgive my ignorance but why do we need clone here

Reply:
For  the reasons above we need to use some kind of smart pointer inside `TxOrdering`. Smart pointers manage data outside of their `size`. On the other hand, and I'm quoting some SO answer here:
> The Copy trait represents values that can be safely duplicated via memcpy

The idea here is, if you have to manage data outside of your `size` you cannot copy it with memcpy, so you cannot have smart pointers implementing `Copy`.

Before my changes to `TxOrdering`, because of the `sort_tx_with_aux_rand` signature (`sort_tx_with_aux_rand(self...`, notice self is owned by the function), and as `params` is later used in `create_tx`, Rust[1] performed a copy to avoid the partial move.

That's not possible anymore as `TxOrdering` cannot implement `Copy` with a smart pointer in its fields, so the new way to avoid the partial move is with `clone()`.

[1] Here I'm guessing based on the suggestion from clippy.

Reply:
Great thanks for clarifying

Reply:
You should be able to remove this `.clone()` by changing the `TxOrdering` functions to borrow `&self` instead of taking ownership of `self`. 
```rust
impl TxOrdering {

    pub fn sort_tx(&self, tx: &mut Transaction) ...
     
    pub fn sort_tx_with_aux_rand(&self, tx: &mut Transaction, rng: &mut impl RngCore) ...
}
```

Reply:
@nymius what do you think about removing this `.clone()` ?

Reply:
Great! I thought there were stronger reasons to take ownership, but If that's not the case, I prefer the borrow.